components/withBreadcrumbs.tsx

```diff
@@ -2,29 +2,40 @@
 
 import type { FC } from 'react';
 
+import type { BreadcrumbLink } from '@/components/Common/Breadcrumbs';
 import Breadcrumbs from '@/components/Common/Breadcrumbs';
 import { useClientContext, useMediaQuery, useSiteNavigation } from '@/hooks';
 import type { NavigationKeys } from '@/types';
+import { dashToCamelCase } from '@/util/stringUtils';
 
 const WithBreadcrumbs: FC = () => {
   const { navigationItems, getSideNavigation } = useSiteNavigation();
   const { pathname } = useClientContext();
-
   const isMobileScreen = useMediaQuery('(max-width: 639px)');
 
   const getBreadrumbs = () => {
     const [navigationKey] =
       navigationItems.find(([, item]) => pathname.includes(item.link)) || [];
-
     if (navigationKey === undefined) {
       return [];
     }
-
-    return getSideNavigation([navigationKey as NavigationKeys])
-      .map(([, item]) => item.items)
-      .flat()
-      .filter(([, item]) => pathname.includes(item.link))
-      .map(([, item]) => ({ label: item.label, href: item.link }));
+    const navigationTree = getSideNavigation([navigationKey as NavigationKeys]);
+    const pathList = pathname
+      .split('/')
+      .filter(item => item !== '')
+      .map(dashToCamelCase);
+
+    let currentNode = navigationTree;
+
+    return pathList.reduce((breadcrumbs, path) => {
```

Comment:
As a last change, can you add some inline code docs explaining what this code is doing, just to make it easy for newcomers to see what it is doing.

Reply:
okok, Thank you for your patient help!